### PR TITLE
SONAR-9072 for api/components/suggestions use all valid search tokens

### PR DESCRIPTION
**Pull request by bartfastiel**
The search for suggestions should ignore tokens, that are only one character long (with warning), but use all other tokens for the search.

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/SuggestionsAction.java [hunk 1]
```diff
@@ -200,6 +202,12 @@ private SuggestionsWsResponse loadSuggestionsWithoutSearch(int skip, int limit,
   }
 
   private SuggestionsWsResponse loadSuggestionsWithSearch(String query, int skip, int limit, Set<String> recentlyBrowsedKeys, List<String> qualifiers) {
+    if (split(query).noneMatch(token -> token.length() >= MINIMUM_NGRAM_LENGTH)) {
+      SuggestionsWsResponse.Builder queryBuilder = newBuilder();
+      getWarning(query).ifPresent(queryBuilder::setWarning);
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/SuggestionsAction.java]: Having a single call to `getWarning()` in this method would be better but ok, it's acceptable.

File: server/sonar-server/src/main/java/org/sonar/server/es/textsearch/ComponentTextSearchFeatureRepertoire.java [hunk 2]
```diff
@@ -53,35 +57,54 @@ public QueryBuilder getQuery(ComponentTextSearchQuery query) {
   },
   PREFIX(CHANGE_ORDER_OF_RESULTS) {
     @Override
-    public QueryBuilder getQuery(ComponentTextSearchQuery query) {
-      return prefixAndPartialQuery(query.getQueryText(), query.getFieldName(), SEARCH_PREFIX_ANALYZER)
+    public Stream<QueryBuilder> getQueries(ComponentTextSearchQuery query) {
+      List<String> tokens = split(query.getQueryText());
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/es/textsearch/ComponentTextSearchFeatureRepertoire.java]: The split is executed multiple times, once per search criteria. It should be part of `ComponentTextSearchQuery`.

File: server/sonar-server/src/main/java/org/sonar/server/es/textsearch/ComponentTextSearchFeatureRepertoire.java [hunk 3]
```diff
@@ -53,35 +57,54 @@ public QueryBuilder getQuery(ComponentTextSearchQuery query) {
...
     }
   },
   KEY(GENERATE_RESULTS) {
     @Override
-    public QueryBuilder getQuery(ComponentTextSearchQuery query) {
-      return matchQuery(SORTABLE_ANALYZER.subField(query.getFieldKey()), query.getQueryText())
-        .boost(50f);
+    public Stream<QueryBuilder> getQueries(ComponentTextSearchQuery query) {
+      String queryText = query.getQueryText();
+      if (ComponentKeys.isValidModuleKey(queryText)) {
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/es/textsearch/ComponentTextSearchFeatureRepertoire.java]: Is it expected to not search for file keys anymore ? Here only projects and modules are supported.

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/SuggestionsAction.java (hunk 1 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/SuggestionsAction.java]: Not trivial to refactor, so I'll leave this for the moment.

File: server/sonar-server/src/main/java/org/sonar/server/es/textsearch/ComponentTextSearchFeatureRepertoire.java (hunk 3 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/es/textsearch/ComponentTextSearchFeatureRepertoire.java]: I've dropped this change.

File: server/sonar-server/src/main/java/org/sonar/server/es/textsearch/ComponentTextSearchFeatureRepertoire.java (hunk 2 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/es/textsearch/ComponentTextSearchFeatureRepertoire.java]: Done.

**Pull request merged**